src/EditorFeatures/CSharpTest/RemoveUnusedMembers/RemoveUnusedMembersTests.cs

```diff
@@ -0,0 +1,1475 @@
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    public void M()
+    {
+        int x;
+        (_goo, x) = (0, 0);
+    }
+}",
+@"class MyClass
+{
+    public void M()
+    {
+        int x;
+        (_goo, x) = (0, 0);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsOnlyWritten_ObjectInitializer()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class MyClass
+{
+    private int [|_goo|];
+    public MyClass M() => new MyClass() { _goo = 0 };
+}",
+@"
+class MyClass
+{
+    public MyClass M() => new MyClass() { _goo = 0 };
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsOnlyWritten_InProperty()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    int Goo
+    {
+        get { return 0; }
+        set { _goo = value; }
+    }
+}",
+@"class MyClass
+{
+    int Goo
+    {
+        get { return 0; }
+        set { _goo = value; }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsReadAndWritten()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    public void M()
+    {
+        _goo = 0;
+        System.Console.WriteLine(_goo);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task PropertyIsReadAndWritten()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|P|] { get; set; }
+    public void M()
+    {
+        P = 0;
+        System.Console.WriteLine(P);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task IndexerIsReadAndWritten()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|this|][int x] { get { return 0; } set { } }
+    public void M(int x)
+    {
+        this[x] = 0;
+        System.Console.WriteLine(this[x]);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsReadAndWritten_InProperty()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    int Goo
+    {
+        get { return _goo; }
+        set { _goo = value; }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsIncremented()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    public int M1() => ++_goo;
```

Comment:
thse are weird cases.  The value matters because it's being returned through M1, not because there is a ++.  For example, say the method was:

```c#
void M1() { ++_goo; }
```

This would still be a candidate for notification since the value was never actually 'read' in a way that is observable.

Reply:
This is now handled in the analyzer.

---
In reply to: [212812599](https://github.com/dotnet/roslyn/pull/29511#discussion_r212812599) [](ancestors = 212812599)